tracer/test/Datadog.Trace.Tests/Ci/Ipc/IpcTests.cs

```diff
@@ -0,0 +1,88 @@
+// <copyright file="IpcTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Ci.Ipc;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Ci.Ipc;
+
+public class IpcTests
+{
+    [Fact]
+    public async Task IpcClientTest()
+    {
+        var name = nameof(IpcClientTest) + "-" + Guid.NewGuid().ToString("n");
+        using var server = new IpcServer(name);
+        using var client = new IpcClient(name);
+
+        TestMessage? finalValue = null;
+        var serverTaskCompletion = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+        var clientTaskCompletion = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        server.MessageReceived += (sender, message) =>
+        {
+            var value = (TestMessage)message;
+            if (value.ServerValue < 100)
+            {
+                value.ServerValue++;
+                while (!server.TrySendMessage(value))
+                {
+                    Thread.Sleep(100);
+                }
+
+                Interlocked.Exchange(ref finalValue, value);
+            }
+
+            if (value.ServerValue == 100)
+            {
+                serverTaskCompletion.TrySetResult(true);
+            }
+        };
+
+        client.MessageReceived += (sender, message) =>
+        {
+            var value = (TestMessage)message;
+            if (value.ClientValue < 100)
+            {
+                value.ClientValue++;
+                while (!client.TrySendMessage(value))
+                {
+                    Thread.Sleep(100);
+                }
+
+                Interlocked.Exchange(ref finalValue, value);
+            }
+
+            if (value.ClientValue == 100)
+            {
+                clientTaskCompletion.TrySetResult(true);
+            }
+        };
+
+        client.TrySendMessage(new TestMessage(0, 0));
+
+        var delayTask = Task.Delay(30_000);
```

Comment:
How long do these tests _Actually_ take on macOS in CI? 😉  Maybe we should reduce the iterations if it's too long